proto/viam/provisioning/provisioning.proto

```diff
@@ -0,0 +1,78 @@
+syntax = "proto3";
+
+package viam.provisioning.v1;
+
+option go_package = "go.viam.com/api/provisioning/v1";
+
+// ProvisioningService is used by viam-agent to retrieve device and smart machine info.
+service ProvisioningService {
+  // GetDeviceStatus is for retrieving the status of the device including networking.
+  rpc GetDeviceStatus(GetDeviceStatusRequest) returns (GetDeviceStatusResponse);
+
+  // SetWifiCredentials is to set the wifi credentials.
+  rpc SetWifiCredentials(SetWifiCredentialsRequest) returns (SetWifiCredentialsResponse);
+
+  // SetSmartMachineCredentials is to set the smart machine credentials.
+  rpc SetSmartMachineCredentials(SetSmartMachineCredentialsRequest) returns (SetSmartMachineCredentialsResponse); 
+
+  // GetNetworkList is to retrieve the list of networks that are visible to the device.
+  rpc GetNetworkList(GetNetworkListRequest) returns (GetNetworkListResponse); 
+}
+
+message GetDeviceStatusRequest {
+}
+
+message GetDeviceStatusResponse {
+  ProviosioningInfo provisioning_info = 1;
+  bool has_device_credentials = 2;
```

Comment:
This either needs to be "smart_machine_credentials" OR we should change the other RPC to be "SetDeviceCredentials" I'm not sure which directly is better. I like the shortness of "device" but it's also very generic.

Reply:
Agree with you, End up going for smart_machine, because device is too generic I think. The ideal short term would be robot 🙈 . I think smart_machine correctly conveys the intent even if it's long.